src/callback-endpoint.ts

```diff
@@ -42,24 +42,35 @@ export const createCallbackEndpoint = (
       // obtain access token
       // /////////////////////////////////////
 
-      const tokenResponse = await fetch(pluginOptions.tokenEndpoint, {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/x-www-form-urlencoded",
-          Accept: "application/json",
-        },
-        body: new URLSearchParams({
-          code,
-          client_id: pluginOptions.clientId,
-          client_secret: pluginOptions.clientSecret,
-          redirect_uri: redirectUri,
-          grant_type: "authorization_code",
-        }).toString(),
-      });
-      const tokenData = await tokenResponse.json();
-      const access_token = tokenData?.access_token;
-      if (typeof access_token !== "string")
-        throw new Error(`No access token: ${JSON.stringify(tokenData)}`);
+      let access_token: string;
+
+      if (pluginOptions.getToken) {
+        access_token = pluginOptions.getToken(code);
+
+        if (typeof access_token !== "string")
+          throw new Error(`No access token: ${access_token}`);
+      } else {
+        const tokenResponse = await fetch(pluginOptions.tokenEndpoint, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/x-www-form-urlencoded",
+            Accept: "application/json",
+          },
+          body: new URLSearchParams({
+            code,
+            client_id: pluginOptions.clientId,
+            client_secret: pluginOptions.clientSecret,
+            redirect_uri: redirectUri,
+            grant_type: "authorization_code",
+          }).toString(),
+        });
+        const tokenData = await tokenResponse.json();
+
+        access_token = tokenData?.access_token;
+
+        if (typeof access_token !== "string")
+          throw new Error(`No access token: ${JSON.stringify(tokenData)}`);
```

Comment:
These 2 lines are duplicated from above, can you move it out of the if else tree?

Reply:
Hello, I wanted to keep logging the full json response in the default way of getting the token. Or we dont need that?

Reply:
not in the default way, that's the point of having a custom `getToken` function

Reply:
Makes sense. Done.